### PR TITLE
Added extra optimized flags support such as avx, IB, mlx etc

### PR DESCRIPTION
**Pull request by amd-toolchain-support**
- Major optional flags have been incorporated. This will improve intra-node communication.

### CODE REVIEW

File: var/spack/repos/builtin/packages/ucx/package.py [hunk 1]
```diff
@@ -60,6 +60,24 @@ class Ucx(AutotoolsPackage, CudaPackage):
             description='Enable KNEM support')
     variant('xpmem', default=False,
             description='Enable XPMEM support')
+    variant('cma', default=False,
+            description="nable Cross Memory Attach")
+    variant('avx', default=False,
+            description="Enable AVX option")
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/ucx/package.py]: Can we avoid a variant and inspect the target instead to make a decision?

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/ucx/package.py]: Unfortunately there is no such generic option available with UCX. There are total five arch related flags, out of these compiler specific flags two flags --with-mcpu and --with-march searches for ThunderX architectures. Two other flags related to SSE. And the fifth flag we have added i.e --with-avx which suits best for latest architectures. 

If you want, we will remove --with-avx flag and update the package. Please suggest how to go about this.

Comment (alalazo) [reply var/spack/repos/builtin/packages/ucx/package.py]: To be clear, for the `avx` variant what I mean is that you can automatically do something like:
```python
if 'avx' in self.spec.target:
    # Do something based on avx detection
```
and drop the variant. This is in line with what we do in other packages, like `gromacs`.

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/ucx/package.py]: Addressed the same. Thanks.

File: var/spack/repos/builtin/packages/ucx/package.py [hunk 2]
```diff
@@ -88,16 +104,32 @@ def configure_args(self):
...
+            config_args.append('--disable-cma')
+
         if '+paramter_checking' in spec:
             config_args.append('--enable-params-check')
         else:
             config_args.append('--disable-params-check')
 
+        # Activate SIMD based on properties of the target
+        if self.spec.target >= 'zen':
+            config_args.append('--with-avx')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/ucx/package.py]: Does this not apply to Intel architectures or piledriver (bdver2) and above?

Comment (amd-toolchain-support) [reply var/spack/repos/builtin/packages/ucx/package.py]: Agree, please have a look at latest code changes.

**Comment by alalazo** [comment]
@hppritcha

**Comment by hppritcha** [comment]
Looks good :+1:

**Review by alalazo — APPROVED** [review]

**Comment by sjpb** [comment]
Setting these defaults to `off` seems unhelpful, we've just spent a while tracking down why my spack build (v0.14.2) of UCX had IB whereas someone else's (v0.16.2) didn't. So I can use them right, can I ask is there a reason you can't just turn them on by default? Presumably if the hardware doesn't support the transport it'll just not work in UCX anyway and then UCX won't auto-select it?

**Comment by amd-toolchain-support** [comment]
Hi @sjpb ,

we adhere the spack community guidelines to develop UCX, if hardware doesn't support certain flags they will be explicitly disabled/not recommended.

Thanks.

**Pull request merged**